docs/_docs/monitoring-metrics/performance-statistics.adoc

```diff
@@ -0,0 +1,85 @@
+// this work for additional information regarding copyright ownership.
+// The ASF licenses this file to You under the Apache License, Version 2.0
+// (the "License"); you may not use this file except in compliance with
+// the License.  You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+= Performance statistics
+
+WARNING: This feature is experimental and can be changed in future releases.
+
+== Overview
+
+Ignite provides a built-in tool for cluster profiling.
+
+You can link:#collecting-statistics[collect] performance statistics from the cluster and then link:#building-the-report[build] the performance report.
+
+== Collecting statistics
+
+To start and stop collecting statistics use the link:#jmx[JMX interface].
+
+Each node collects performance statistics to the file in binary format. This file is placed under `Ignite_work_directory/perf_stat/` directory.
+The name mask is `node-{nodeId}-{index}.prf`.
+
+Performance statistics files are used to build the report offline. Stop collecting statistics before creating the report.
+
+Nodes use off-heap buffer to temporarily store serialized statistics. The writer thread flushes buffer to the file when the flush size is reached. See the link:#system-properties[properties] section for details.
+
+Each start of collecting statistics will create a new file on nodes. The index in the file name will be increased if such file exists:
+
+    node-faedc6c9-3542-4610-ae10-4ff7e0600000.prf
+    node-faedc6c9-3542-4610-ae10-4ff7e0600000-1.prf
+    node-faedc6c9-3542-4610-ae10-4ff7e0600000-2.prf
+
+== Building the report
+
+Ignite provides a tool to generate the report from performance statistics files. The tool is published in the ignite-extensions repository: https://github.com/apache/ignite-extensions/tree/master/modules/performance-statistics-ext/[performance-statistics-ext, windows="_blank"].
+
+To build the performance report follow:
+
+1. Collect performance statistics files from all nodes under an empty directory. For example:
+
+    path_to_files
+        ├── node-162c7147-fef8-4ea2-bd25-8653c41fc7fa.prf
+        ├── node-7b8a7c5c-f3b7-46c3-90da-e66103c00001.prf
+        └── node-faedc6c9-3542-4610-ae10-4ff7e0600000.prf
+
+2.  Run the script from the release package of the tool:
+
+        performance-statistics-tool/build-report.sh path_to_files
+
+The performance report will be created in the new directory under the performance statistics files path:
+`path_to_files/report_yyyy-MM-dd_HH-mm-ss/`. Open `report_yyyy-MM-dd_HH-mm-ss/index.html` in the browser to see the report.
+
+== Management
+
```

Comment:
You should not leave the empty line between two headings. Put something in there, for example:
'Here you'll learn about JMX interface and system properties.'

Reply:
Fixed.